crates/router/src/core/payments/operations/payment_create.rs

```diff
@@ -694,6 +694,10 @@ impl<F: Clone + Send> Domain<F, api::PaymentsRequest, PaymentData<F>> for Paymen
                 }),
                 payment_method_type: None,
             });
+            payment_data
```

Comment:
we should not update payment attempt when calculating the tax, since this operation is not dependent on payment attempt. @swangi-kumari can you please confirm this once?

Reply:
Yes ideally it shouldn't depend on payment attempt but in v1, we are storing tax calculation details in net_amount of payment_attempt. In v2, we can have different way to handle this

Reply:
There is separate field storing this data in intent itself, @swangi-kumari can you confirm this once?

Reply:
We store tax_details in payment_intent
```
pub struct PaymentIntent {
    pub tax_details: Option<TaxDetails>,
}
```


```
pub struct TaxDetails {
    pub default: Option<DefaultTax>,
    pub payment_method_type: Option<PaymentMethodTypeTax>,
}
```

Reply:
Yes we store tax_details in payment_intent, but we would need to update payment_attempt.net_amount with order_tax_amount too so that all the subsequent flows would use the updated net_amount which includes order_tax_amount

Reply:
After confirm only we need to compute total_amount using intent, for create the related information will be stored on intent and the computation also on intent.